src/ripple/app/tests/TxQ_test.cpp

```diff
@@ -360,6 +360,205 @@ class TxQ_test : public TestSuite
             metrics.txPerLedger,
             256, lastMedian);
     }
+
+    void testLastLedgerSeq()
+    {
+        using namespace jtx;
+
+        Env env(*this, makeConfig());
+
+        auto& txq = env.app().getTxQ();
+        txq.setMinimumTx(2);
+
+        auto alice = Account("alice");
+        auto bob = Account("bob");
+        auto charlie = Account("charlie");
+        auto daria = Account("daria");
+        auto edgar = Account("edgar");
+        auto felicia = Account("felicia");
+
+        auto queued = ter(terQUEUED);
+
+        checkMetrics(env, 0, boost::none, 0, 2, 256, 500);
+
+        // Fund these accounts and close the ledger without
+        // involving the queue, so that stats aren't affected.
+        env.fund(XRP(1000), noripple(alice, bob, charlie, daria, edgar, felicia));
+        env.close();
+
+        checkMetrics(env, 0, boost::none, 0, 2, 256, 500);
+        submit(env,
+            env.jt(noop(bob)));
+        submit(env,
+            env.jt(noop(charlie)));
+        submit(env,
+            env.jt(noop(daria)));
+        checkMetrics(env, 0, boost::none, 3, 2, 256, 500);
+
+        // Queue an item with a LastLedgerSeq.
+        submit(env,
+            env.jt(noop(alice), json("{\"LastLedgerSequence\":4}"),
```

Comment:
Could use a raw string here to get rid of the backslashes: `R"("{"LastLedgerSequence":4})"`